### ISSUE TITLE
Log file /tmp/PT_fBkCmo is empty. This means a PHPUnit process has crashed.

### ISSUE THREAD

**Issue by lartie** [paratestphp/paratest#397]
Hello, I have a problem.

```
~/.composer/vendor/bin/paratest --verbose=1 -c `pwd`/phpunit.xml --phpunit `pwd`/vendor/bin/phpunit -p 1

Running phpunit in 1 process with /var/www/app/vendor/bin/phpunit

Configuration read from /var/www/app/phpunit.xml


Executing test via: /usr/local/bin/php Array Array
An error for 1: Log file /tmp/PT_fBkCmo is empty. This means a PHPUnit process has crashed.
The process: /usr/local/bin/php Array Array
This means a PHPUnit process was unable to run "/var/www/app/src/Core/Tests/CoreTest.php"

Command: /usr/local/bin/php Array Array
StdErr:
StdOut: Could not open input file: Array
```

```
~/.composer/vendor/bin/paratest --version
ParaTest 2.2.0
```

```
vendor/bin/phpunit --version
PHPUnit 7.5.13-2-g1477fe424 by Sebastian Bergmann and contributors.
```

```
php --version
PHP 7.1.27 (cli) (built: Mar  9 2019 03:49:52) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2018 Zend Technologies
    with Xdebug v2.7.0, Copyright (c) 2002-2019, by Derick Rethans
```

**Comment by denizsokullu** [comment]
I have a similar issue when running -f to log --coverage-php and I am doing it on 36 processes. Is it a problem to have 36 processes? Does the File IO become a bottleneck for the tests?

**Comment by crzdeveloper** [comment]
IMHO, the CLI is strange: `Command: /usr/local/bin/php Array Array`. It seems, an array is passed as the argument of the PHP. Could you please provide the contents of `/var/www/app/phpunit.xml`? Please also try updating the paratest

**Comment by lartie** [comment]
@crzdeveloper 

I've tried this configuration (from readme)

```
<?xml version="1.0" encoding="UTF-8"?>
<phpunit backupGlobals="false"
         backupStaticAttributes="false"
         bootstrap="../bootstrap.php"
         colors="true"
         convertErrorsToExceptions="true"
         convertNoticesToExceptions="true"
         convertWarningsToExceptions="true"
         processIsolation="false"
         stopOnFailure="false"
         syntaxCheck="false"
        >
    <testsuites>
        <testsuite name="ParaTest Fixtures">
            <directory>./tests/</directory>
        </testsuite>
    </testsuites>
</phpunit>
```

I can't update the paratest because I have to use the PHP 7.1.* version.

**Comment by Slamdunk** [comment]
Hi, could you please test it with the latest version, currently `4.1.0`?

**Comment by designvoid** [comment]
Any update on this? Facing same issue ☹️

**Comment by Slamdunk** [comment]
Current version `5.0` gives thorough details about the error.

What's up with you, @designvoid ?

**Comment by designvoid** [comment]
Ahh, I'm version locked at the mo through other deps - I will put a pin in this and check again once I'm fully up to date! Cheers for the info @Slamdunk

**Comment by EvKoh** [comment]
I have same error...


`
Cannot load Xdebug - it was already loaded
Running phpunit in 8 processes with /var/www/project/vendor/phpunit/phpunit/phpunit

Configuration read from /var/www/project/phpunit.xml.dist


In Runner.php line 136:
                                                                                                                                                                                                                  
  [ParaTest\Runners\PHPUnit\WorkerCrashedException]                                                                                                                                                               
  The command "'/usr/local/bin/php' '/var/www/project/vendor/phpunit/phpunit/phpunit' '--configuration' '/var/www/project/phpunit.xml.dist' '--log-junit' '/tmp/PT_KlioFK' '/var/www/project/tests/Functional/Au  
  th/ JwtAuthTest.php'" failed.                                                                                                                                                                                   
                                                                                                                                                                                                                  
  Exit Code: 1(General error)                                                                                                                                                                                     
                                                                                                                                                                                                                  
  Working directory: /var/www/project                                                                                                                                                                             
                                                                                                                                                                                                                  
  Output:                                                                                                                                                                                                         
  ================                                                                                                                                                                                                
  Class ' JwtAuthTest' could not be found in '/var/www/project/tests/Functional/Auth/ JwtAuthTest.php'.                                                                                                           
                                                                                                                                                                                                                  
                                                                                                                                                                                                                  
  Error Output:                                                                                                                                                                                                   
  ================                                                                                                                                                                                                
  Cannot load Xdebug - it was already loaded                                                                                                                                                                      
                                                                                                                                                                                                                  

Exception trace:
  at /var/www/project/vendor/brianium/paratest/src/Runners/PHPUnit/Runner.php:136
 ParaTest\Runners\PHPUnit\Runner->testIsStillRunning() at /var/www/project/vendor/brianium/paratest/src/Runners/PHPUnit/Runner.php:45
 ParaTest\Runners\PHPUnit\Runner->doRun() at /var/www/project/vendor/brianium/paratest/src/Runners/PHPUnit/BaseRunner.php:77
 ParaTest\Runners\PHPUnit\BaseRunner->run() at /var/www/project/vendor/brianium/paratest/src/Console/Commands/ParaTestCommand.php:86
 ParaTest\Console\Commands\ParaTestCommand->execute() at /var/www/project/vendor/symfony/console/Command/Command.php:258
 Symfony\Component\Console\Command\Command->run() at /var/www/project/vendor/symfony/console/Application.php:920
 Symfony\Component\Console\Application->doRunCommand() at /var/www/project/vendor/symfony/console/Application.php:266
 Symfony\Component\Console\Application->doRun() at /var/www/project/vendor/symfony/console/Application.php:142
 Symfony\Component\Console\Application->run() at /var/www/project/vendor/brianium/paratest/bin/paratest:37

In ResultPrinter.php line 187:
                                                                                                                                                                                                                  
  [ParaTest\Runners\PHPUnit\EmptyLogFileException]                                                                                                                                                                
  Log file /tmp/PT_KlioFK is empty. This means a PHPUnit process has crashed.                                                                                                                                     
  The process: '/usr/local/bin/php' '/var/www/project/vendor/phpunit/phpunit/phpunit' '--configuration' '/var/www/project/phpunit.xml.dist' '--log-junit' '/tmp/PT_KlioFK' '/var/www/project/tests/Functional/Au  
  th/ JwtAuthTest.php'                                                                                                                                                                                            
  This means a PHPUnit process was unable to run "/var/www/project/tests/Functional/Auth/ JwtAuthTest.php"                                                                                                        
                                                                                                                                                                                                                  

Exception trace:
  at /var/www/project/vendor/brianium/paratest/src/Runners/PHPUnit/ResultPrinter.php:187
 ParaTest\Runners\PHPUnit\ResultPrinter->printFeedback() at /var/www/project/vendor/brianium/paratest/src/Runners/PHPUnit/Runner.php:134
 ParaTest\Runners\PHPUnit\Runner->testIsStillRunning() at /var/www/project/vendor/brianium/paratest/src/Runners/PHPUnit/Runner.php:45
 ParaTest\Runners\PHPUnit\Runner->doRun() at /var/www/project/vendor/brianium/paratest/src/Runners/PHPUnit/BaseRunner.php:77
 ParaTest\Runners\PHPUnit\BaseRunner->run() at /var/www/project/vendor/brianium/paratest/src/Console/Commands/ParaTestCommand.php:86
 ParaTest\Console\Commands\ParaTestCommand->execute() at /var/www/project/vendor/symfony/console/Command/Command.php:258
 Symfony\Component\Console\Command\Command->run() at /var/www/project/vendor/symfony/console/Application.php:920
 Symfony\Component\Console\Application->doRunCommand() at /var/www/project/vendor/symfony/console/Application.php:266
 Symfony\Component\Console\Application->doRun() at /var/www/project/vendor/symfony/console/Application.php:142
 Symfony\Component\Console\Application->run() at /var/www/project/vendor/brianium/paratest/bin/paratest:37

In Reader.php line 45:
                                                                               
  [InvalidArgumentException]                                                   
  Log file /tmp/PT_KlioFK is empty. This means a PHPUnit process has crashed.  
                                                                               

Exception trace:
  at /var/www/project/vendor/brianium/paratest/src/Logging/JUnit/Reader.php:45
 ParaTest\Logging\JUnit\Reader->__construct() at /var/www/project/vendor/brianium/paratest/src/Runners/PHPUnit/ResultPrinter.php:185
 ParaTest\Runners\PHPUnit\ResultPrinter->printFeedback() at /var/www/project/vendor/brianium/paratest/src/Runners/PHPUnit/Runner.php:134
 ParaTest\Runners\PHPUnit\Runner->testIsStillRunning() at /var/www/project/vendor/brianium/paratest/src/Runners/PHPUnit/Runner.php:45
 ParaTest\Runners\PHPUnit\Runner->doRun() at /var/www/project/vendor/brianium/paratest/src/Runners/PHPUnit/BaseRunner.php:77
 ParaTest\Runners\PHPUnit\BaseRunner->run() at /var/www/project/vendor/brianium/paratest/src/Console/Commands/ParaTestCommand.php:86
 ParaTest\Console\Commands\ParaTestCommand->execute() at /var/www/project/vendor/symfony/console/Command/Command.php:258
 Symfony\Component\Console\Command\Command->run() at /var/www/project/vendor/symfony/console/Application.php:920
 Symfony\Component\Console\Application->doRunCommand() at /var/www/project/vendor/symfony/console/Application.php:266
 Symfony\Component\Console\Application->doRun() at /var/www/project/vendor/symfony/console/Application.php:142
 Symfony\Component\Console\Application->run() at /var/www/project/vendor/brianium/paratest/bin/paratest:37

paratest [--bootstrap BOOTSTRAP] [--colors] [-c|--configuration CONFIGURATION] [--coverage-clover COVERAGE-CLOVER] [--coverage-crap4j COVERAGE-CRAP4J] [--coverage-html COVERAGE-HTML] [--coverage-php COVERAGE-PHP] [--coverage-test-limit COVERAGE-TEST-LIMIT] [--coverage-text] [--coverage-xml COVERAGE-XML] [--exclude-group EXCLUDE-GROUP] [--filter FILTER] [-f|--functional] [-g|--group GROUP] [-h|--help] [--log-junit LOG-JUNIT] [-m|--max-batch-size MAX-BATCH-SIZE] [--no-test-tokens] [--parallel-suite] [--passthru PASSTHRU] [--passthru-php PASSTHRU-PHP] [--path PATH] [--phpunit PHPUNIT] [-p|--processes PROCESSES] [--runner RUNNER] [--stop-on-failure] [--testsuite TESTSUITE] [--tmp-dir TMP-DIR] [-v|--verbose VERBOSE] [--whitelist WHITELIST] [--] [<path>]

+ set +x

`

**Comment by Slamdunk** [comment]
> `Class ' JwtAuthTest' could not be found in '/var/www/project/tests/Functional/Auth/ JwtAuthTest.php'.`

The space at start of class name is weird: can you open a new Issue for that, and maybe add more details about your SUT?

**Comment by filipagh** [comment]
@Slamdunk after update phpunit 8.5 -> 9.1 i got same error

<details>
  <summary>Log  paratest 4.2 </summary>
> 
> Running phpunit in 3 processes with /srv/http/LiveAgent/LiveAgent/server/vendor/phpunit/phpunit/phpunit
> 
> Configuration read from /srv/http/LiveAgent/LiveAgent/server/phpunit.xml
> 
> ..............
> In ResultPrinter.php line 182:
>                                                                                
>   Log file /tmp/PT_52m7FQ is empty. This means a PHPUnit process has crashed.  
>   The process: '/usr/bin/php' '/srv/http/LiveAgent/LiveAgent/server/vendor/ph  
>   punit/phpunit/phpunit' '--configuration' '/srv/http/LiveAgent/LiveAgent/ser  
>   ver/phpunit.xml' '--log-junit' '/tmp/PT_52m7FQ' '/srv/http/LiveAgent/LiveAg  
>   ent/server/include/Gpf/Tests/ApplicationTest.class.php'                      
>   This means a PHPUnit process was unable to run "/srv/http/LiveAgent/LiveAge  
>   nt/server/include/Gpf/Tests/ApplicationTest.class.php"                                                                                     

</details>



<details>
  <summary>Log paratest 5.0</summary>
Running phpunit in 3 processes with /srv/http/LiveAgent/LiveAgent/server/vendor/phpunit/phpunit/phpunit

Configuration read from /srv/http/LiveAgent/LiveAgent/server/phpunit.xml

..........
In Runner.php line 136:
                                                                               
  The command "'/usr/bin/php' '/srv/http/LiveAgent/LiveAgent/server/vendor/ph  
  punit/phpunit/phpunit' '--configuration' '/srv/http/LiveAgent/LiveAgent/ser  
  ver/phpunit.xml' '--log-junit' '/tmp/PT_aqRArH' '/srv/http/LiveAgent/LiveAg  
  ent/server/include/Gpf/Tests/File/UploadBaseTest.class.php'" failed.         
                                                                               
  Exit Code: 1(General error)                                                  
                                                                               
  Working directory: /srv/http/LiveAgent/LiveAgent/server                      
                                                                               
  Output:                                                                      
  ================                                                             
  Class 'UploadBaseTest.class' could not be found in '/srv/http/LiveAgent/Liv  
  eAgent/server/include/Gpf/Tests/File/UploadBaseTest.class.php'.              
                                                                             
In ResultPrinter.php line 187:
                                                                               
  Log file /tmp/PT_aqRArH is empty. This means a PHPUnit process has crashed.  
  The process: '/usr/bin/php' '/srv/http/LiveAgent/LiveAgent/server/vendor/ph  
  punit/phpunit/phpunit' '--configuration' '/srv/http/LiveAgent/LiveAgent/ser  
  ver/phpunit.xml' '--log-junit' '/tmp/PT_aqRArH' '/srv/http/LiveAgent/LiveAg  
  ent/server/include/Gpf/Tests/File/UploadBaseTest.class.php'                  
  This means a PHPUnit process was unable to run "/srv/http/LiveAgent/LiveAge  
  nt/server/include/Gpf/Tests/File/UploadBaseTest.class.php"                   
                                                                               

In Reader.php line 45:
                                                                               
  Log file /tmp/PT_aqRArH is empty. This means a PHPUnit process has crashed.  

</details>                                                                       
<br/><br/>
env:
PHP 7.4.1<br/>
PS. paratest 4* and phpunit 8.5 works OK

**Comment by Slamdunk** [comment]
> Class 'UploadBaseTest.class' could not be found in '/srv/http/LiveAgent/LiveAgent/server/include/Gpf/Tests/File/UploadBaseTest.class.php'.

Only PSR-4 class naming is supported right now, have a look at its standards: https://www.php-fig.org/psr/psr-4/

**Comment by filipagh** [comment]
yes we are working on rewrite all to psr-4 standard but we have lot of legacy tests
phpunit support also non psr4 test class  (using --test-suffix and test dir)

actual paratest cli cmd:

>   '/usr/bin/php' '/srv/http/LiveAgent/LiveAgent/server/vendor/phpunit/phpunit/phpunit' '--configuration' '/srv/http/LiveAgent/LiveAgent/server/phpunit.xml' '--log-junit' '/tmp/PT_tNfUlB544' '/srv/http/LiveAgent/LiveAgent/server/include/Gpf/Tests/ApplicationTest.class.php'

is possible to change cmd builder? to:

>   '/usr/bin/php' '/srv/http/LiveAgent/LiveAgent/server/vendor/phpunit/phpunit/phpunit' '--configuration' '/srv/http/LiveAgent/LiveAgent/server/phpunit.xml' **--test-suffix ApplicationTest.class.php** '--log-junit' '/tmp/PT_tNfUlB544' **'/srv/http/LiveAgent/LiveAgent/server/include/Gpf/Tests'**

base on phpunit.xml conf 

```
<testsuites>
    <testsuite name="Unit">
      <directory suffix="Test.php">./tests/Qu/La</directory>
      <directory suffix="Test.class.php">./include/Gpf/Tests</directory>
      <directory suffix="Test.class.php">./include/La/Tests</directory>
    </testsuite>
...
```

**Comment by Slamdunk** [comment]
> is possible to change cmd builder?

You can try with the `--passthru` option.

**Comment by filipagh** [comment]
i need also change test file -> dir (where test is) (last arg)

**Comment by gjm** [comment]
I have similar issue but only happens when I run `paratest` in a Gitlab CI/CD pipeline.

**Comment by Slamdunk** [comment]
Hi, from version 5 the error on crash has been made verbose and debuggable.

Please open a new specific issue with the extended stack trace if you need non-generic support.

**Comment by gjm** [comment]
@Slamdunk how can I obtain the debug? Just add the `--verbose` option and post the exception trace? Is that enough?

**Comment by Slamdunk** [comment]
Yes

**Comment by MajorasJack** [comment]
Bumping this thread up, still having this on every build running via CircleCI & AWS CodeBuild, works fine locally (all instances running PHP 7.4 with the same ini config

Command being run: `./vendor/bin/paratest -p 8 --coverage-clover deployment/coverage.xml --verbose`

Stacktrace of the error with the `--verbose` flag added:

`In CoverageMerger.php line 59:
                                                                               
  [ParaTest\Coverage\EmptyCoverageFileException]                               
  Coverage file /tmp/CV_L8DgkT is empty. This means a PHPUnit process has cra  
  shed.                                                                        
                                                                               

Exception trace:
  at /root/project/vendor/brianium/paratest/src/Coverage/CoverageMerger.php:59
 ParaTest\Coverage\CoverageMerger->addCoverageFromFile() at /root/project/vendor/brianium/paratest/src/Runners/PHPUnit/Runner.php:118
 ParaTest\Runners\PHPUnit\Runner->testIsStillRunning() at /root/project/vendor/brianium/paratest/src/Runners/PHPUnit/Runner.php:43
 ParaTest\Runners\PHPUnit\Runner->doRun() at /root/project/vendor/brianium/paratest/src/Runners/PHPUnit/BaseRunner.php:81
 ParaTest\Runners\PHPUnit\BaseRunner->run() at /root/project/vendor/brianium/paratest/src/Console/Commands/ParaTestCommand.php:84
 ParaTest\Console\Commands\ParaTestCommand->execute() at /root/project/vendor/symfony/console/Command/Command.php:256
 Symfony\Component\Console\Command\Command->run() at /root/project/vendor/symfony/console/Application.php:971
 Symfony\Component\Console\Application->doRunCommand() at /root/project/vendor/symfony/console/Application.php:290
 Symfony\Component\Console\Application->doRun() at /root/project/vendor/symfony/console/Application.php:166
 Symfony\Component\Console\Application->run() at /root/project/vendor/brianium/paratest/bin/paratest:37`

**Comment by eytienne** [comment]
Same here, no idea why, no output given:

```
unning phpunit in 8 processes with /<CUT>/vendor/phpunit/phpunit/phpunit. Functional mode is ON.

Configuration read from /<CUT>/phpunit.xml


In WorkerCrashedException.php line 36:
                                                                                                                                                                                                          
  The command "'/usr/local/bin/php' '/<CUT>/vendor/phpunit/phpunit/phpunit' '--configuration' '/<CUT>/phpunit.xml' '--filter' '/<CUT>(?:\s|$)/' '--no-logging' '--no-coverage' '--printer  
  ' 'ParaTest\Runners\PHPUnit\Worker\NullPhpunitPrinter' '--log-junit' '/tmp/PT_GDPlDJ' '/<CUT>/test/<CUT>.php'" failed.                                                                         
                                                                                                                                                                                                          
  Exit Code: 0(OK)                                                                                                                                                                                        
                                                                                                                                                                                                          
  Working directory: /legalife                                                                                                                                                                            
                                                                                                                                                                                                          
  Output:                                                                                                                                                                                                 
  ================                                                                                                                                                                                        
                                                                                                                                                                                                          
                                                                                                                                                                                                          
  Error Output:                                                                                                                                                                                           
  ================                                                                                                                                                                                        
                                                                                                                                                                                                          

In ResultPrinter.php line 227:
                                                                                                                                                                                                          
  Log file /tmp/PT_GDPlDJ is empty. This means a PHPUnit process has crashed.                                                                                                                             
  The process: '/usr/local/bin/php' '/<CUT>/vendor/phpunit/phpunit/phpunit' '--configuration' '/<CUT>/phpunit.xml' '--filter' '/<CUT>(?:\s|$)/' '--no-logging' '--no-coverage' '--printer  
  ' 'ParaTest\Runners\PHPUnit\Worker\NullPhpunitPrinter' '--log-junit' '/tmp/PT_GDPlDJ' '/<CUT>/test/<CUT>.php'                                                                                  
  This means a PHPUnit process was unable to run "/legalife/test/<CUT>.php"                                                                                                                         
                                                                                                                                                                                                          

In Reader.php line 45:
                                                                               
  Log file /tmp/PT_GDPlDJ is empty. This means a PHPUnit process has crashed.

**Comment by eytienne** [comment]
Same here, no idea why, no output given:

```
unning phpunit in 8 processes with /<CUT>/vendor/phpunit/phpunit/phpunit. Functional mode is ON.

Configuration read from /<CUT>/phpunit.xml


In WorkerCrashedException.php line 36:
                                                                                                                                                                                                          
  The command "'/usr/local/bin/php' '/<CUT>/vendor/phpunit/phpunit/phpunit' '--configuration' '/<CUT>/phpunit.xml' '--filter' '/<CUT>(?:\s|$)/' '--no-logging' '--no-coverage' '--printer  
  ' 'ParaTest\Runners\PHPUnit\Worker\NullPhpunitPrinter' '--log-junit' '/tmp/PT_GDPlDJ' '/<CUT>/test/<CUT>.php'" failed.                                                                         
                                                                                                                                                                                                          
  Exit Code: 0(OK)                                                                                                                                                                                        
                                                                                                                                                                                                          
  Working directory: /<CUT>                                                                                                                                                                            
                                                                                                                                                                                                          
  Output:                                                                                                                                                                                                 
  ================                                                                                                                                                                                        
                                                                                                                                                                                                          
                                                                                                                                                                                                          
  Error Output:                                                                                                                                                                                           
  ================                                                                                                                                                                                        
                                                                                                                                                                                                          

In ResultPrinter.php line 227:
                                                                                                                                                                                                          
  Log file /tmp/PT_GDPlDJ is empty. This means a PHPUnit process has crashed.                                                                                                                             
  The process: '/usr/local/bin/php' '/<CUT>/vendor/phpunit/phpunit/phpunit' '--configuration' '/<CUT>/phpunit.xml' '--filter' '/<CUT>(?:\s|$)/' '--no-logging' '--no-coverage' '--printer  
  ' 'ParaTest\Runners\PHPUnit\Worker\NullPhpunitPrinter' '--log-junit' '/tmp/PT_GDPlDJ' '/<CUT>/test/<CUT>.php'                                                                                  
  This means a PHPUnit process was unable to run "/<CUT>/test/<CUT>.php"                                                                                                                         
                                                                                                                                                                                                          

In Reader.php line 45:
                                                                               
  Log file /tmp/PT_GDPlDJ is empty. This means a PHPUnit process has crashed.  
                                                                               
```

**Comment by MajorasJack** [comment]
@eytienne as this issue is now closed, please can you move your comment to #614
